src/kernel-install/kernel-install

```diff
@@ -90,7 +90,15 @@ if [[ ! $COMMAND ]] || [[ ! $KERNEL_VERSION ]]; then
     exit 1
 fi
 
-[ -z "$MACHINE_ID" ] && [ -f /etc/machine-id ] && [ "$(stat -fc %T /etc/machine-id)" != "tmpfs" ] && read -r MACHINE_ID < /etc/machine-id
+# Prefer to use an existing machine ID from /etc/machine-info or /etc/machine-id. If we're using the machine
+# ID /etc/machine-id, try to persist it in /etc/machine-info. If no machine ID is found, try to generate
+# a new machine ID in /etc/machine-info. If that fails, use "Default".
+
+[ -z "$MACHINE_ID" ] && [ -f /etc/machine-info ] && source /etc/machine-info && MACHINE_ID="$KERNEL_INSTALL_MACHINE_ID"
+[ -z "$MACHINE_ID" ] && [ -f /etc/machine-id ] && read -r MACHINE_ID </etc/machine-id
+[ -n "$MACHINE_ID" ] && [ -z "$KERNEL_INSTALL_MACHINE_ID" ] && echo "KERNEL_INSTALL_MACHINE_ID=$MACHINE_ID" >>/etc/machine-info
+[ -z "$MACHINE_ID" ] && echo "KERNEL_INSTALL_MACHINE_ID=$(systemd-id128 new)" >>/etc/machine-info
+[ -z "$MACHINE_ID" ] && [ -f /etc/machine-info ] && source /etc/machine-info && MACHINE_ID="$KERNEL_INSTALL_MACHINE_ID"
 [ -z "$MACHINE_ID" ] && MACHINE_ID="Default"
```

Comment:
Shouldn't this be inverted now, i.e. just KERNEL_INSTALL_MACHINE_ID set? It's confusing to set MACHINE_ID, even if it may not match the actual machine-id.

Reply:
Github seems to be acting up again, which line does this comment apply to?

Reply:
To the big block that sets MACHINE_ID in various way.

Reply:
I kept MACHINE_ID since it's still public API so we have to check for it regardless. We need a second variable aside from KERNEL_INSTALL_MACHINE_ID so we can check if /etc/machine-info already contains KERNEL_INSTALL_MACHINE_ID.